src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1758,132 +1756,53 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserWithParams(
-                            }
-                            StudentAttributes student = null;
-                            Set<String> studentsEmailInTeam = null;
-                            boolean isVisibleResponse = isResponseVisibleForUser(userEmail,
-                                    role, student, studentsEmailInTeam, response,
-                                    question, instructor);
-                            if (isVisibleResponse) {
-                                responses.add(response);
-                                addEmailNamePairsToTable(emailNameTable, response,
-                                        question, roster);
-                                addEmailLastNamePairsToTable(emailLastNameTable, response,
-                                        question, roster);
-                                addEmailTeamNamePairsToTable(emailTeamNameTable,
-                                        response,
-                                        question, roster);
-                                addVisibilityToTable(visibilityTable, question,
-                                        response, userEmail, role, roster);
-                            }
-                        }
-                    }
-                }
-            }
 
-            addSectionTeamNamesToTable(sectionTeamNameTable, roster, courseId, userEmail, role,
-                                       feedbackSessionName, section);
-            
-            FeedbackSessionResultsBundle results =
-                    new FeedbackSessionResultsBundle(
-                            session, responses, relevantQuestions, emailNameTable,
-                            emailLastNameTable, emailTeamNameTable, sectionTeamNameTable,
-                            visibilityTable, responseStatus, roster, responseComments, true);
+        String section = params.get(PARAM_SECTION);
+        String questionId = params.get(PARAM_QUESTION_ID);
 
-            return results;
+        if (questionId != null) {
+            return getFeedbackSessionResultsForQuestionId(feedbackSessionName, courseId, userEmail, role, roster,
+                    session, relevantQuestions, allQuestions, isIncludeResponseStatus, section, questionId);
         }
 
+        // create empty data containers to store results
+        List<FeedbackResponseAttributes> responses = new ArrayList<FeedbackResponseAttributes>();
+
+        Map<String, String> emailNameTable = new HashMap<String, String>();
+        Map<String, String> emailLastNameTable = new HashMap<String, String>();
+        Map<String, String> emailTeamNameTable = new HashMap<String, String>();
+        Map<String, Set<String>> sectionTeamNameTable = new HashMap<String, Set<String>>();
+        Map<String, boolean[]> visibilityTable = new HashMap<String, boolean[]>();
+
+
         Map<String, FeedbackQuestionAttributes> allQuestionsMap = new HashMap<String, FeedbackQuestionAttributes>();
```

Comment:
This reminds me of the for all question block that we see above, duplicate or warranted?